std/stdio.d

```diff
@@ -1075,27 +1075,53 @@ Throws: `ErrnoException` if the file is not opened or if the call to `fwrite` fa
 
         version (Windows)
         {
-            flush(); // before changing translation mode
             immutable fd = ._fileno(_p.handle);
-            immutable mode = ._setmode(fd, _O_BINARY);
-            scope(exit) ._setmode(fd, mode);
+            immutable oldMode = ._setmode(fd, _O_BINARY);
+
             version (DIGITAL_MARS_STDIO)
             {
-                import core.atomic : atomicOp;
-
-                // https://issues.dlang.org/show_bug.cgi?id=4243
                 immutable info = __fhnd_info[fd];
-                atomicOp!"&="(__fhnd_info[fd], ~FHND_TEXT);
-                scope(exit) __fhnd_info[fd] = info;
             }
-            scope(exit) flush(); // before restoring translation mode
```

Comment:
You can still use scope guards throughout here. Just prefix them with the proper `if`. e.g.:

```D
scope(exit) if (oldMode != _O_BINARY) flush();
```

Same with the setting of the mode, and the storing of the info back.

also restore the whole `version (DIGITAL_MARS_STDIO)` block as it was, and put it after the if statement below.

Reply:
Am I to assume to that the atomic operation performed in that version block doesn't have a relatively discernible performance penalty?

Reply:
It's just 1. easier to follow if you put everything into one version block, and 2. easier to prove that it does the same thing as before (your ordering of calls was different than the original).